Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,422 @@
+            headers : collection._headers,
+            asynchronous : collection._asynchronous,
+            debugShowBoundingVolume : collection._debugShowBoundingVolume,
+            debugWireframe : collection._debugWireframe,
+            allowPicking : collection._allowPicking
+        };
+
+        if (collection._instancingEnabled) {
+            var typedArray = new Float32Array(instanceCount * 16);
+            for (i = 0; i < instanceCount; ++i) {
+                var instance = collection._instances[i];
+                var modelMatrix = instance.modelMatrix;
+                var pickColor = defined(instance.pickId) ? instance.pickId.color : Color.WHITE;
+
+                // First three rows of the model matrix
+                typedArray[i * 16 + 0] = modelMatrix[0];
+                typedArray[i * 16 + 1] = modelMatrix[4];
+                typedArray[i * 16 + 2] = modelMatrix[8];
+                typedArray[i * 16 + 3] = modelMatrix[12];
+                typedArray[i * 16 + 4] = modelMatrix[1];
+                typedArray[i * 16 + 5] = modelMatrix[5];
+                typedArray[i * 16 + 6] = modelMatrix[9];
+                typedArray[i * 16 + 7] = modelMatrix[13];
+                typedArray[i * 16 + 8] = modelMatrix[2];
+                typedArray[i * 16 + 9] = modelMatrix[6];
+                typedArray[i * 16 + 10] = modelMatrix[10];
+                typedArray[i * 16 + 11] = modelMatrix[14];
+
+                // Other instance data like pickColor, color, etc. Colors can be packed like in BillboardCollection as needed.
+                typedArray[i * 16 + 12] = pickColor.red;
+                typedArray[i * 16 + 13] = pickColor.green;
+                typedArray[i * 16 + 14] = pickColor.blue;
+                typedArray[i * 16 + 15] = pickColor.alpha;
+            }
+
+            var vertexBuffer = Buffer.createVertexBuffer({
+                context : context,
+                typedArray : typedArray,
+                usage : WebGLConstants.STATIC_DRAW
+            });
+
+            var attributeNames = ['czm_modelMatrixRow0', 'czm_modelMatrixRow1', 'czm_modelMatrixRow2', 'czm_instanceData'];
+            var componentSizeInBytes = ComponentDatatype.getSizeInBytes(ComponentDatatype.FLOAT);
+            var instancedAttributes = {};
+
+            for (i = 0; i < 4; ++i) {
+                instancedAttributes[attributeNames[i]] = {
+                    index                  : 0, // updated in Model
+                    vertexBuffer           : vertexBuffer,
+                    componentsPerAttribute : 4,
+                    componentDatatype      : ComponentDatatype.FLOAT,
+                    normalize              : false,
+                    offsetInBytes          : componentSizeInBytes * 4 * i,
+                    strideInBytes          : componentSizeInBytes * 16,
+                    instanceDivisor        : 1
+                };
+            }
+
+            modelOptions.instanced = true;
```

Comment:
We generally don't dynamically add properties to objects.  Initialize these above to `false` and `undefined`.